geopmdpy/geopmdpy/pio.py

```diff
@@ -593,22 +593,9 @@ def enable_fixed_counters():
     requires the fixed counters to be enabled.
 
     """
-    write_control('MSR::PERF_GLOBAL_CTRL:EN_FIXED_CTR0', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN0_OS', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN0_USR', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN0_PMI', 'board', 0, 0)
-    write_control('MSR::PERF_GLOBAL_CTRL:EN_FIXED_CTR1', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN1_OS', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN1_USR', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN1_PMI', 'board', 0, 0)
-    write_control('MSR::PERF_GLOBAL_CTRL:EN_FIXED_CTR2', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN2_OS', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN2_USR', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN2_PMI', 'board', 0, 0)
-    write_control('MSR::PERF_GLOBAL_CTRL:EN_FIXED_CTR3', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN3_OS', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN3_USR', 'board', 0, 1)
-    write_control('MSR::FIXED_CTR_CTRL:EN3_PMI', 'board', 0, 0)
-    write_control('MSR::PERF_GLOBAL_OVF_CTRL:CLEAR_OVF_FIXED_CTR0', 'board', 0, 0)
-    write_control('MSR::PERF_GLOBAL_OVF_CTRL:CLEAR_OVF_FIXED_CTR1', 'board', 0, 0)
-    write_control('MSR::PERF_GLOBAL_OVF_CTRL:CLEAR_OVF_FIXED_CTR2', 'board', 0, 0)
+    for idx in range(3):
```

Comment:
```suggestion
    for idx in range(4):
```

```
>>> list(range(3))
[0, 1, 2]
```

Reply:
You'll need to add some handling for ``MSR::PERF_GLOBAL_OVF_CTRL:CLEAR_OVF_FIXED_CTR`` on the last iteration since CTR3 doesn't exist.